### PR TITLE
Bugfix/fix example eq zero issue

### LINKED ISSUE #1071: @example 0 does not work
## Sorting

- **I'm submitting a ...**

  - [X] bug report
  - [ ] feature request
  - [ ] support request

- I confirm that I
  - [X] used the [search](https://github.com/lukeautry/tsoa/search?type=Issues) to make sure that a similar issue hasn't already been submit

## Expected Behavior

```json
			"SkipNumber": {
				"type": "integer",
				"format": "int32",
                                "example": 0,
			},
```

```ts
/**
 * @isInt
 * @example 0
 */
export type SkipNumber = number;
```

## Current Behavior

```json 
			"SkipNumber": {
				"type": "integer",
				"format": "int32",
                                "example": 0,
			},
```

```ts

```

## Possible Solution

I think there might be a 
```
https://github.com/lukeautry/tsoa/blob/7c05b4d3f67c313752d38e0efc6826f64b4f27fe/packages/cli/src/swagger/specGenerator3.ts#L165
if (0) evaluates to false.

## Steps to Reproduce

<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->

Create type with example 0.

## Context (Environment)

Version of the library: 3.9.0
Version of NodeJS: 12

- Confirm you were using yarn not npm: [X]

## Detailed Description

<!--- Provide a detailed description of the change or addition you are proposing -->

## Breaking change?

<!--- if this is a breaking change, then please describe how it breaks the existing libraries inputs/outputs -->
<!--- if this is a breaking change, then discuss how we can mitigate that -->

### PR DESCRIPTION
**Pull request by jackneer**
### All Submissions:

- [V] Have you followed the guidelines in our [Contributing](https://github.com/lukeautry/tsoa/tree/master/docs/CONTRIBUTING.md) document?
- [V] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/lukeautry/tsoa/pulls) for the same update/change?
- [V] Have you written unit tests?
- [V] Have you written unit tests that cover the negative cases (i.e.: if bad data is submitted, does the library respond properly)?
- [V] This PR is associated with an existing issue?

**Closing issues**

Put `closes #1071` (where XXXX is the issue number) in your comment to auto-close the issue that your PR fixes.

### If this is a new feature submission:

- [ ] Has the issue had a maintainer respond to the issue and clarify that the feature is something that aligns with the [goals](https://github.com/lukeautry/tsoa#goal) and [philosophy](https://github.com/lukeautry/tsoa#philosophy) of the project?

**Potential Problems With The Approach**

**Test plan**

testing for example = 0, value 0 should be included in the example array

this PR is for [issue](https://github.com/lukeautry/tsoa/issues/1071)

### CODE REVIEW

File: packages/cli/src/swagger/specGenerator2.ts [hunk 1]
```diff
@@ -87,7 +87,7 @@ export class SpecGenerator2 extends SpecGenerator {
           definitions[referenceType.refName].additionalProperties = this.determineImplicitAdditionalPropertiesValue();
         }
 
-        if (referenceType.example) {
+        if (typeof referenceType.example !== 'undefined') {
```
Comment (WoH) [inline packages/cli/src/swagger/specGenerator2.ts:90]:
```suggestion
        if (referenceType.example !== undefined) {
```

File: packages/cli/src/swagger/specGenerator3.ts [hunk 2]
```diff
@@ -162,7 +162,7 @@ export class SpecGenerator3 extends SpecGenerator {
           schema[referenceType.refName].additionalProperties = this.determineImplicitAdditionalPropertiesValue();
         }
 
-        if (referenceType.example) {
+        if (typeof referenceType.example !== 'undefined') {
```
Comment (WoH) [inline packages/cli/src/swagger/specGenerator3.ts:165]:
```suggestion
        if (referenceType.example !== undefined) {
```

File: tests/unit/swagger/definitionsGeneration/metadata.spec.ts [hunk 3]
```diff
@@ -758,6 +758,19 @@ describe('Metadata generation', () => {
       const deprecatedParam2 = method.parameters[2];
       expect(deprecatedParam2.deprecated).to.be.true;
     });
+
+    it('should generate example of 0 for parameter ', () => {
```
Comment (WoH) [inline tests/unit/swagger/definitionsGeneration/metadata.spec.ts:762]: Adding a metadata spec kinda works around what you're trying to show here. This also passes without any code changes. The bug was the metadata -> openapi transformation, not the user code ->  metadata transformation which occurs before. Have a look at the `schemaDetails` and `schemaDetails3` test infra instead.

**Comment by Dancia** [comment]
This is still not merged.

**Pull request closed without merging**